static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -0,0 +1,58 @@
+import {computed, makeObservable} from 'mobx';
+
+import {RawSpanType, SpanChildrenLookupType, SpanType} from './types';
+import {getSpanID} from './utils';
+
+class SpanTreeModel {
+  // readonly state
+  span: SpanType;
+  children: Array<SpanTreeModel> = [];
+
+  constructor(parentSpan: SpanType, childSpans: SpanChildrenLookupType) {
+    this.span = parentSpan;
+
+    const spanID = getSpanID(parentSpan);
+    const spanChildren: Array<RawSpanType> = childSpans?.[spanID] ?? [];
+
+    // Mark descendents as being rendered. This is to address potential recursion issues due to malformed data.
+    // For example if a span has a span_id that's identical to its parent_span_id.
+    childSpans = {
+      ...childSpans,
+    };
+    delete childSpans[spanID];
```

Comment:
Unsure if it makes the intent any clearer but can we use `lodash/omit` here?

Reply:
I didn't use `lodash/omit` here because this is simpler than lodash's implementation: https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L13496